xen/arch/arm/domain.c

```diff
@@ -643,6 +644,7 @@ int arch_sanitise_domain_config(struct xen_domctl_createdomain *config)
     }
 
     if ( config->arch.tee_type != XEN_DOMCTL_CONFIG_TEE_NONE &&
+         config->arch.tee_type != XEN_DOMCTL_CONFIG_TEE_FFA &&
```

Comment:
You should have a look into arch/arm/tee/tee.c as it would be more logical to setup ffa as a tee mediator.
This would also mean to move ffa.c into tee directory an dwould prevent most of the changes in the arm common code.
The only change that i can see is the fact that you will need to add a tee_domain_destroy but that would make sense as there is already a tee_domain_init

Reply:
`ffa_handle_call()` and `tee_handle_call()` use different SMC ID ranges, so we'd still need the changes in `handle_sssc()`. Is that OK?

Reply:
Yes i think there is not other choice for that part. I currently have no idea how to make that part of the code cleaner.
Maybe during upstream review someone will have a better idea.